src/expr/interpret/eval.hpp

```diff
@@ -385,7 +407,7 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
             typename TReturnType = std::enable_if_t<std::is_same_v<TTag, StorageTag>, bool>>
   TReturnType HasLabelImpl(const VertexAccessor &vertex, const LabelIx &label, StorageTag /*tag*/) {
     auto has_label = vertex.HasLabel(view_, GetLabel(label));
-    if (has_label.HasError() && has_label.GetError() == Error::NONEXISTENT_OBJECT) {
+    if (has_label.HasError() && has_label.GetError().code == Error::NONEXISTENT_OBJECT) {
```

Comment:
Maybe you can define the `operator==` for a `ShardError` and `ErrorCode`, so the `.code` part doesn't have to be written out. Of course that means the `ShardError` should be also compared only based on `code` and not the rest. That is questionable move, but it make sense in my opinion, as comparing `ShardError`s is a weird thing to do.

Reply:
ah I already did that, just forgot to use it everywhere